internal/metrics/interceptors.go

```diff
@@ -0,0 +1,147 @@
+			Aggregation: view.Distribution(
+				1, 2, 5, 7, 10, 25, 500, 750,
+				100, 250, 500, 750,
+				1000, 2500, 5000, 7500,
+				10000, 25000, 50000, 75000,
+				100000,
+			),
+		},
+		{
+			Name:        grpcServerResponseSize.Name(),
+			Measure:     grpcServerResponseSize,
+			Description: grpcServerResponseSize.Description(),
+			TagKeys:     []tag.Key{keyService, keyHost, keyMethod, keyStatus, keyGRPCService},
+			Aggregation: view.Distribution(
+				1, 256, 512, 1024, 2048, 8192, 16384, 32768, 65536, 131072, 262144, 524288,
+				1048576, 2097152, 4194304, 8388608,
+			),
+		},
+
+		//grpc Client
+		{
+			Name:        grpcClientRequestCount.Name(),
+			Measure:     grpcClientRequestCount,
+			Description: grpcClientRequestCount.Description(),
+			TagKeys:     []tag.Key{keyService, keyHost, keyMethod, keyStatus, keyGRPCService},
+			Aggregation: view.Count(),
+		},
+		{
+			Name:        grpcClientRequestDuration.Name(),
+			Measure:     grpcClientRequestDuration,
+			Description: grpcClientRequestDuration.Description(),
+			TagKeys:     []tag.Key{keyService, keyHost, keyMethod, keyStatus, keyGRPCService},
+			Aggregation: view.Distribution(
+				1, 2, 5, 7, 10, 25, 500, 750,
+				100, 250, 500, 750,
+				1000, 2500, 5000, 7500,
+				10000, 25000, 50000, 75000,
+				100000,
+			),
+		},
+		{
+			Name:        grpcClientResponseSize.Name(),
+			Measure:     grpcClientResponseSize,
+			Description: grpcClientResponseSize.Description(),
+			TagKeys:     []tag.Key{keyService, keyHost, keyMethod, keyStatus, keyGRPCService},
+			Aggregation: view.Distribution(
+				1, 256, 512, 1024, 2048, 8192, 16384, 32768, 65536, 131072, 262144, 524288,
+				1048576, 2097152, 4194304, 8388608,
+			),
+		},
+	}
+)
+
+func init() {
+	view.Register(grpcViews...)
+}
+
+// GRPCClientInterceptor creates a UnaryClientInterceptor which tracks metrics of grpc client requests
+func GRPCClientInterceptor(service string) grpc.UnaryClientInterceptor {
+	return func(
+		ctx context.Context,
+		method string,
+		req interface{},
+		reply interface{},
+		cc *grpc.ClientConn,
+		invoker grpc.UnaryInvoker,
+		opts ...grpc.CallOption) error {
+
+		startTime := time.Now()
+
+		// Calls the invoker to execute RPC
+		err := invoker(ctx, method, req, reply, cc, opts...)
+
+		// Split the method into parts for better slicing
+		rpcInfo := strings.SplitN(method, "/", 3)
+		var rpcMethod string
+		var rpcService string
+		if len(rpcInfo) == 3 {
+			rpcService = rpcInfo[1]
+			rpcMethod = rpcInfo[2]
+		}
+
+		responseStatus, _ := status.FromError(err)
+		ctx, tagErr := tag.New(
+			context.Background(),
+			tag.Insert(keyService, service),
+			tag.Insert(keyHost, cc.Target()),
+			tag.Insert(keyMethod, rpcMethod),
+			tag.Insert(keyGRPCService, rpcService),
+			tag.Insert(keyStatus, responseStatus.Code().String()),
+		)
+
+		responseProto := reply.(proto.Message)
+		responseSize := len([]byte(responseProto.String()))
```

Comment:
Equivalent to `https://godoc.org/github.com/golang/protobuf/proto#Size` ?

Reply:
Ah, yes.  That's a better way.